AdminCore.FsmWorkflow/WorkflowFsmHandler.cs

```diff
@@ -0,0 +1,135 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using AdminCore.Common;
+using AdminCore.Common.Exceptions;
+using AdminCore.Constants.Enums;
+using AdminCore.DAL;
+using AdminCore.DAL.Models;
+using AdminCore.DTOs.Employee;
+using AdminCore.DTOs.Event;
+using AdminCore.FsmWorkflow.Factory;
+using AdminCore.FsmWorkflow.FsmMachines;
+using AdminCore.FsmWorkflow.FsmMachines.FsmLeaveStates;
+using AdminCore.FsmWorkflow.FsmMachines.FsmWorkflowState;
+
+namespace AdminCore.FsmWorkflow
+{
+    public class WorkflowFsmHandler : IWorkflowFsmHandler
+    {
+        private IDatabaseContext _dbContext;
+        private IWorkflowFsmFactory<ILeaveWorkflow> _workflowFsmFactory;
+        public WorkflowFsmHandler(IDatabaseContext dbContext, IWorkflowFsmFactory<ILeaveWorkflow> workflowFsmFactory)
+        {
+            _dbContext = dbContext;
+            _workflowFsmFactory = workflowFsmFactory;
+        }
+        
+        public EventWorkflow CreateEventWorkflow(int eventTypeId, bool saveChangesToDbContext)
+        {
+            var eventWorkflow = new EventWorkflow
+            {
+                WorkflowState = GetInitialWorkflowState(eventTypeId)
+            };
+            
+            _dbContext.EventWorkflowRepository.Insert(eventWorkflow);
+            if (saveChangesToDbContext)
+            {
+                _dbContext.SaveChanges();
+            }
+
+            return eventWorkflow;
+        }
+
+        public WorkflowFsmStateInfo FireLeaveResponse(EventDto employeeEvent, EmployeeDto respondeeEmployee, EventStatuses eventStatus, EventWorkflow eventWorkflow)
+        {
+            ILeaveWorkflow workflowFsm;
+            var workflowStateData = RebuildWorkflowStateData(eventWorkflow);
+            
+            switch (employeeEvent.EventTypeId)
+            {
+                case (int)EventTypes.AnnualLeave:
+                    workflowFsm = _workflowFsmFactory.GetWorkflowPto(workflowStateData);
+                    break;
+                case (int)EventTypes.WorkingFromHome:
+                    workflowFsm = _workflowFsmFactory.GetWorkflowWfh(workflowStateData);
+                    break;
+                default:
+                    throw new WorkflowException($"No workflow FSM exists for {((EventTypes)employeeEvent.EventTypeId).ToString()}");
+            }
+            
+            // Cast is not redundant, need the int value of enum as a string
+            // ReSharper disable once RedundantCast
```

Comment:
what's this about?

Reply:
Resharper was flagging this as a redundant cast, the "// ReSharper disable once RedundantCast" comment suppresses that warning in Resharper.

Resharper is right as it usually is, I originally was casting to an enum so it was once not redundant. Removing these lines.